x-pack/plugins/lens/public/drag_drop/drag_drop.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, useContext } from 'react';
+import { DragContext } from './providers';
+
+type DroppableEvent = React.DragEvent<HTMLElement>;
+
+/**
+ * A function that handles a drop event.
+ */
+type DropHandler = (item: any) => void;
```

Comment:
I like either using `unknown` or saying they at least have to have a signature like `{ type: 'string' }` so that they can be somewhat systematically reasoned about.